plugin/build.gradle.kts

```diff
@@ -82,6 +131,27 @@ gradlePlugin {
     }
 }
 
+// Need to move publishing configuration into afterEvaluate {}
+// to override changes done by "com.gradle.plugin-publish" plugin in afterEvaluate {} block
+// See PublishPlugin class for details
+afterEvaluate {
+    publishing {
+        publications {
+            withType(MavenPublication::class.java) {
+                // Special workaround to publish shadow jar instead of normal one. Name to override peeked here:
+                // https://github.com/gradle/gradle/blob/master/subprojects/plugin-development/src/main/java/org/gradle/plugin/devel/plugins/MavenPluginPublishPlugin.java#L73
+                if (name == "pluginMaven") {
+                    setArtifacts(
+                        listOf(
+                            shadowJarTask.get()
+                        )
+                    )
+                }
+            }
+        }
+    }
+}
```

Comment:
Can you verify that the `pom.xml` that we're generating doesn't have these shadowed dependencies?

If there's a way to verify with a build task that we aren't adding any unexpected additional dependencies, that would be useful.

Also, something like this may help us out too for sanity checking.
```
val ensureDependenciesAreInlined by tasks.registering {
    dependsOn(tasks.shadowJar)
    doLast {
        zipTree(tasks.shadowJar.flatMap { it.archiveFile }).visit {
            if (!isDirectory) {
                val path = relativePath
                if (path.lastName.endsWith(".class") && !path.pathString.startsWith(pluginGroup.replace(".", "/")) {
                    throw GradleException("Found non inlined dependency: " + path.pathString)
                }
            }
        }
    }
}
```

Reply:
By running `publishToMavenLocal` I see that neither `pom` file nor gradle metadata do not contain dependencies. 

That was the reason I've introduced `shadowImplementation` configuration - is to remove `implementation` dependencies to be added to generated pom/gradle metadata, but still shadow them into final jar file.

I've also tried to update to snapshot version in my work project - works ok.

You task looks interesting, will see if it works :+1:

Reply:
I've added proposed task in https://github.com/JLLeitschuh/ktlint-gradle/pull/323/commits/c56aa4551e027407a1b38220d917de4e6389b5c1 commit